Source/CombatExtended/CombatExtended/Verbs/Verb_MeleeAttackCE.cs

```diff
@@ -756,15 +756,21 @@ protected virtual void DoParry(Pawn defender, Thing parryThing, bool isRiposte =
             }
             else
             {
-                Verb_MeleeAttackCE verb = defender.meleeVerbs.TryGetMeleeVerb(caster) as Verb_MeleeAttackCE;
-                if (verb == null)
+                Verb verb = defender.meleeVerbs.TryGetMeleeVerb(caster);
+                Verb_MeleeAttackCE ceVerb = verb as Verb_MeleeAttackCE;
+                if (verb is Verb_MeleeApplyHediff hediffVerb)
+                {
+                    hediffVerb.ApplyMeleeDamageToTarget(caster);
+                    sound = hediffVerb.SoundHitPawn();
+                }
+                if (ceVerb == null)
```

Comment:
won't we still hit this branch for the `verb is Verb_MeleeApplyHediff hediffVerb` case? we might need to wrap it into an else